samples/bluetooth/peripheral_hr_coded/src/main.c

```diff
@@ -0,0 +1,191 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/printk.h>
+#include <zephyr.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/services/bas.h>
+#include <bluetooth/services/hrs.h>
+
+static struct k_work start_advertising_worker;
+
+static struct bt_le_ext_adv *adv;
+static struct bt_conn *default_conn;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_UUID16_ALL, 0x0d, 0x18, 0x0f, 0x18, 0x0a, 0x18),
+};
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	int err;
+	struct bt_conn_info info;
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		printk("Connection failed (err %d)\n", conn_err);
+		return;
+	}
+
+	default_conn = bt_conn_ref(conn);
+
+	err = bt_conn_get_info(conn, &info);
+
+	if (err) {
+		printk("Failed to get connection info\n");
+	} else {
+		const struct bt_conn_le_phy_info *phy_info;
+		phy_info = info.le.phy;
+
+		printk("Connected: %s, tx_phy %u, rx_phy %u\n",
+				addr, phy_info->tx_phy, phy_info->rx_phy);
+	}
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	printk("Disconnected (reason 0x%02x)\n", reason);
+
+	if (default_conn) {
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+	}
+
+	k_work_submit(&start_advertising_worker);
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static int create_advertising_coded(void)
+{
+	int err;
+	struct bt_le_adv_param param = {
+		.id           = BT_ID_DEFAULT,
+		.sid          = 0,
+		.interval_min = BT_GAP_ADV_FAST_INT_MIN_2,
+		.interval_max = BT_GAP_ADV_FAST_INT_MAX_2,
+		.options      = BT_LE_ADV_OPT_CONNECTABLE
+		                | BT_LE_ADV_OPT_EXT_ADV
+		                | BT_LE_ADV_OPT_CODED
```

Comment:
@joerchan Did your changes, and also changed the initialization of `struct bt_le_adv_param param`